enterprise/views.py

```diff
@@ -539,6 +563,51 @@ def get_enterprise_course_enrollment_page(self, request, enterprise_uuid, course
         return render_to_response('enterprise_course_enrollment_page.html', context_data, request=request)
 
     @method_decorator(login_required)
+    def post(self, request, enterprise_uuid, course_id):
+        """
+        Process a submitted track selection form for the enterprise.
+        """
+        enterprise_customer, course, modes = self.get_base_details(enterprise_uuid, course_id)
+
+        selected_mode = request.POST.get('course_mode')
+
+        matching_modes = [mode for mode in modes if mode['mode'] == selected_mode]
+
+        if not matching_modes:
+            return self.get_enterprise_course_enrollment_page(request, enterprise_customer, course, modes)
+        else:
+            if matching_modes[0].get('premium', False):
+                # Premium modes should be directed to the ecommerce flow
+                return redirect(
+                    reverse(
+                        'verify_student_start_flow',
```

Comment:
Can we please avoid using the reverse urls defined in `LMS` and manually generate url? As in future we are expecting `edx-enterprise` to become a separate IDA.